flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java

```diff
@@ -221,6 +221,22 @@
 	/** Defines the configuration key of that external resource in Kubernetes. This is used as a suffix in an actual config. */
 	public static final String EXTERNAL_RESOURCE_KUBERNETES_CONFIG_KEY_SUFFIX = "kubernetes.config-key";
 
+	public static final ConfigOption<Map<String, String>> KUBERNETES_SECRETS =
+		key("kubernetes.secrets")
+			.mapType()
+			.noDefaultValue()
+			.withDescription("The user-specified secrets that are mounted into Flink container. The value should be in " +
+				"the form of a1:v1,a2:v2. For example for mounting secrets foo to path /opt/secrets , just need to set " +
+				"kubernetes.secrets: foo:/opt/secrets in the flink-conf.yaml");
+
+	public static final ConfigOption<List<Map<String, String>>> KUBERNETES_ENV_SECRET_KEY_REF =
+		key("kubernetes.env.secretKeyRef")
+			.mapType()
+			.asList()
+			.noDefaultValue()
+			.withDescription("the user-specified secrets to set env variable in Flink container. The value should be in " +
+				"the form of env:MY_ENV1,secret:my_secret1,key:my_key1;env:MY_ENV2,secret:my_secret2,key:my_key2");
```

Comment:
env:MY_ENV1,secret:my_secret1,key:my_key1;env:MY_ENV2,secret:my_secret2,key:my_key2

I prefer not using **my** in the description. So it could be like following.
env:FOO_ENV,secret:foo_secret,key:key;env:BAR_ENV,secret:bar_secret,key:key

Reply:
fixed